core/renderers/common/renderer.js

```diff
@@ -49,10 +49,16 @@ Blockly.blockRendering.Renderer = function(name) {
 /**
  * Initialize the renderer.
  * @param {!Blockly.Theme} theme The workspace theme object.
+ * @param {Object=} opt_rendererOverrides Rendering constant overrides.
  * @package
  */
-Blockly.blockRendering.Renderer.prototype.init = function(theme) {
+Blockly.blockRendering.Renderer.prototype.init = function(theme,
+    opt_rendererOverrides) {
   this.constants_ = this.makeConstants_();
+  if (opt_rendererOverrides) {
+    this.constants_.overrides = opt_rendererOverrides;
```

Comment:
You're setting this directly from out here instead of changing `makeConstants_`.  Is that to avoid/minimize API changes?

Reply:
And if so, is it actually helping?  The developer still needs to modify their override of `renderer.init` and `renderer.refresh`, and now  needs to remember to add the overrides in the correct order.

Reply:
Yes

Reply:
To your second comment, init is our entry point from the workspace, and we don't have other any other option for passing renderer overrides. 

I don't imagine overriding the init method would be common, but if a developer does (as we do in geras), I would expect a super call. 
Moving this logic to ``makeConstants_`` would have the same problem if not worse, since its a factory method for creating the constants provider and that is way more common of an override.

Reply:
Reasoning sounds good.

LGTM to this PR